src/prefect/flow_runs.py

```diff
@@ -114,17 +118,37 @@ async def main(num_runs: int):
 
             ```
     """
+    if poll_interval is not None:
+        get_logger().warning(
+            "The `poll_interval` argument is deprecated and will be removed in a future release. "
+        )
+
     assert client is not None, "Client injection failed"
     logger = get_logger()
+
+    filter = EventFilter(
+        event=EventNameFilter(prefix=["prefect.flow-run"]),
+        resource=EventResourceFilter(id=[f"prefect.flow-run.{flow_run_id}"])
+    )
+
     with anyio.move_on_after(timeout):
-        while True:
+        async with get_events_subscriber(filter=filter) as subscriber:
```

Comment:
consequence of above
```suggestion
        async with get_events_subscriber(filter=event_filter) as subscriber:
```